src/redteam/commands/run.ts

```diff
@@ -0,0 +1,127 @@
+  const configPath = options.config || 'promptfooconfig.yaml';
+  const redteamPath = options.output || 'redteam.yaml';
+
+  // Check if promptfooconfig.yaml exists, if not, run init
+  if (!fs.existsSync(configPath)) {
+    logger.info('No configuration file found. Running initialization...');
+    await redteamInit(undefined);
+    // User probably needs to edit init and stuff, so it is premature to generate and eval.
+    return;
+  }
+
+  // Check for updates to the config file and regenerate redteam if necessary
+  let shouldGenerate = true;
+  if (fs.existsSync(redteamPath)) {
+    const redteamContent = yaml.load(fs.readFileSync(redteamPath, 'utf8')) as any;
+    const storedHash = redteamContent.metadata?.configHash;
+    const currentHash = getConfigHash(configPath);
+
+    if (storedHash === currentHash) {
+      shouldGenerate = false;
+    }
+  }
+
+  if (shouldGenerate) {
+    logger.info('Generating new test cases...');
+    await doGenerateRedteam({
+      ...options,
+      config: configPath,
+      output: redteamPath,
+    } as Partial<RedteamCliGenerateOptions>);
+
+    // Update redteam.yaml with the new config hash
+    const redteamContent = yaml.load(fs.readFileSync(redteamPath, 'utf8')) as any;
+    redteamContent.metadata = {
+      ...redteamContent.metadata,
+      configHash: getConfigHash(configPath),
+    };
+    fs.writeFileSync(redteamPath, yaml.dump(redteamContent));
+  } else {
+    logger.info('Using existing test cases...');
+  }
+
+  // Run evaluation
+  const { defaultConfig } = await loadDefaultConfig();
+  await doEval(
+    {
+      ...options,
+      config: [redteamPath],
+    } as Partial<CommandLineOptions & Command>,
+    defaultConfig,
+    redteamPath,
+    {
+      showProgressBar: true,
+    },
+  );
+
+  logger.info(chalk.green('\nRed team evaluation complete!'));
+  logger.info(chalk.blue('To view the results, run: ') + chalk.bold('promptfoo redteam report'));
+}
+
+export function runRedteamCommand(program: Command) {
+  program
+    .command('run')
+    .description('Run red teaming process (init, generate, and evaluate)')
+    .option('-c, --config [path]', 'Path to configuration file. Defaults to promptfooconfig.yaml')
+    .option(
+      '-o, --output [path]',
+      'Path to output file for generated tests. Defaults to redteam.yaml',
+    )
+    .option('--no-cache', 'Do not read or write results to disk cache', false)
+    .option('--env-file, --env-path <path>', 'Path to .env file')
+    .option('-j, --max-concurrency <number>', 'Maximum number of concurrent API calls', (val) =>
+      Number.parseInt(val, 10),
+    )
+    .option('--delay <number>', 'Delay in milliseconds between API calls', (val) =>
+      Number.parseInt(val, 10),
+    )
+    .option('--remote', 'Force remote inference wherever possible', false)
+    .action(async (opts: RedteamRunOptions) => {
+      try {
+        if (opts.remote) {
+          cliState.remote = true;
+        }
+        await doRedteamRun(opts);
```

Comment:
Process exit 0 after it's complete?

Reply:
that is the default exit code, so I don't think it's necessary?